swift/StableDiffusion/pipeline/StableDiffusionXL+Resources.swift

```diff
@@ -76,10 +80,12 @@ public extension StableDiffusionXLPipeline {
 
         // Refiner Unet model
         let unetRefiner: Unet?
-        if FileManager.default.fileExists(atPath: urls.unetRefinerURL.path) {
-            unetRefiner = Unet(modelAt: urls.unetRefinerURL, configuration: config)
+        if FileManager.default.fileExists(atPath: urls.unetRefinerChunk1URL.path) &&
+            FileManager.default.fileExists(atPath: urls.unetRefinerChunk2URL.path) {
+            unetRefiner = Unet(chunksAt: [urls.unetRefinerChunk1URL, urls.unetRefinerChunk2URL],
+                               configuration: config)
         } else {
-            unetRefiner = nil
+            unetRefiner = Unet(modelAt: urls.unetRefinerURL, configuration: config)
```

Comment:
Shouldn't `unetRefiner` be `nil` when it's not there?

Reply:
Update: after testing I can confirm that the generation loop fails if there's no refiner in the bundle.

Reply:
Yes, good catch. I copied this from the controlnet logic but missed the first parts.